File: A2/A2-017/A2-017.py
PRICE = {'R':{'S':60, 'M':80, 'L':100},
         'T':{'S':80, 'M':100, 'L':120}}
TOPPING = {'P':15, 'E':10}

# Input ramen information
ramen_size, ramen_type = input().strip().split()
price = PRICE[ramen_type][ramen_size]

# Input topping information
topping = input().strip().split()
if(len(topping) == 2):
    topping_type, topping_amount = topping[0], int(topping[1])
    price += TOPPING[topping_type] * topping_amount

# Output
print(price)

File: A2/A2-017/A2-017.cpp
#include<iostream>
using namespace std;

const int PRICE[2][3] = {{60, 80, 100}, {80, 100, 120}};
const int TOPPING[2] = {15, 10};
char ramenType, ramenSize, toppingType;
int type01, type02, toppingAmount, price = 0;

int main()
{
    // This makes std::cin and std::cout faster
    ios_base::sync_with_stdio(false); cin.tie(0);
    
    // Input ramen information
    cin >> ramenSize >> ramenType;
    if(ramenType == 'R') { type01 = 0; }
    if(ramenType == 'T') { type01 = 1; }
    if(ramenSize == 'S') { type02 = 0; }
    if(ramenSize == 'M') { type02 = 1; }
    if(ramenSize == 'L') { type02 = 2; }
    price += PRICE[type01][type02];

    // Input topping information
    cin >> toppingType;
    if(toppingType == 'P')
    {
        cin >> toppingAmount;
        price += TOPPING[0] * toppingAmount;
    }
    if(toppingType == 'E')
    {
        cin >> toppingAmount;
        price += TOPPING[1] * toppingAmount;
    }

    // Output
    cout << price;
    return 0;
}